lesson13/script.js

```diff
@@ -0,0 +1,83 @@
+function showLoadingImg(){
+    const div = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id = "js-loading";
+    loadingImg.src = "img/loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function fetchdata(){
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    //responseがokではないとき
+    if(!response.ok){
+    throw new Error(`${response.status}:${response.text}`);
+    }
+    //responseがokのとき
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+}
+
+function renderList(result) {
+    const fragment = document.createDocumentFragment();
+    const ul = document.getElementById('js-list');
+        for(const list of result){
+            const li = document.createElement('li');
+            const a = document.createElement('a');
+            const img = document.createElement('img');
+            a.textContent = list.text;
+            a.href = `/${list.a}`;
+            img.src = list.img;
+            img.alt = list.alt;
+            fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+        }
+    ul.appendChild(fragment);
+}
+
+async function init(){
+    showLoadingImg();
+    try{
+        const result = await fetchdata();
+        //resultがtrueではないとき
+        if (!result) {
+            return;
+        }
+        //resultがtrueのとき
+        renderList(result);
+    }catch(e){
+        document.getElementById('js-list').textContent = e;
+        console.error(e);
```

Comment:
I will share what I learned from my teacher's review :)

> The 'e' would look like an EVENT object, so it should be clear that it is an ERROR object.

Reply:
Thank you for teaching!

Usually, catch is an error object, so the name "e" seems to be common.

However, if I want to throw an exception myself, it is not necessarily an Error object, as I can specify any value.
Therefore, it seems that I need to give it a name that you know will cause an error.

I will  rename to error!

Reply:
I change the variable name of the formal argument of catch.
https://github.com/nakagawamai/js-lesson/pull/15/commits/1feb3e7bd05133ee5efa673892055ca0a4f5887b